### DISCUSSION TITLE
Reminders for only when a habit is due

### DISCUSSION THREAD

**Discussion by navinthomas** [iSoron/uhabits#1342] [Feature Requests] [answered]
### Feature
A user should be able to set the reminder to appear only when the habit is due.

### Use Case
If a non-daily habit's frequency is, for example, set to once in three days, then setting reminders at designated days of the week is not sufficient. The app should be able to show a reminder once 3 days has elapsed since the last time that habit was done.

Example:  On a Tuesday, a user create a habit "Water plants" with frequency set to once every three days. User marks the habit as done on that day (Tuesday). The user needs to do the habit only on Friday. The reminder should be displayed on Friday. If the user does not mark the habit as done on that day, then the reminder should pop up again the next day, and the next day, until the user marks it has completed. Then the reminder is shown again only after 3 days.

Such a feature needs a more dynamic/flexible reminder option than what is available on the app at the moment.

**Comment by iSoron** [comment, marked as answer]
> The app should be able to show a reminder once 3 days has elapsed since the last time that habit was done.

This is already the current app behavior. If there is an implicit checkmark for the habit, no reminders are shown. This is explained in the FAQ:

> Loop will only show you a reminder if the task has not been already completed that day. For example, if your reminder is scheduled to show up at 8am, but you already added a checkmark manually (on the main app) at 7am, then the reminder will not appear. The same is true for non-daily habits. If you have a weekly habit, and you complete it on a certain Monday, the reminder will only appear on the following Monday.

**Reply by navinthomas** [reply]
Oh. That means I should enable daily reminders for even non-daily habits, right? So, the app is configured to show a reminder on any day of the week. But, if the habit has already been done as per the defined frequency, then the app won't show the reminder until the habits needs to be done again (as per the defined frequency).

So, the reminder configuration is more like an indication of which days of the week the app is allowed to show the reminder, not days of the week it should explicitly show the reminder.

Correct me if I am wrong. I am anyway going to test it out to see if my understanding is correct.

**Reply by iSoron** [reply]
That's right, @navinthomas. The reminder configuration is just to prevent the app, for example, from showing work-related reminders during the weekend just because you couldn't complete them during the week. Perhaps it would be more clear if we had labelled that as "Disallow reminders on: Saturday, Sunday".
